tests/Integration/Behaviour/Features/Context/Domain/CategoryFeatureContext.php

```diff
@@ -245,101 +228,214 @@ public function categoryDoesNotExist(string $categoryReference)
-        $command->setIsActive($editableRootCategoryTestCaseData->isActive());
-        $command->setLocalizedLinkRewrites($editableRootCategoryTestCaseData->getLinkRewrite());
-        $command->setLocalizedNames($editableRootCategoryTestCaseData->getName());
-        $command->setLocalizedDescriptions($editableRootCategoryTestCaseData->getDescription());
-        $command->setLocalizedAdditionalDescriptions($editableRootCategoryTestCaseData->getAdditionalDescription());
-        $command->setLocalizedMetaTitles($editableRootCategoryTestCaseData->getMetaTitle());
-        $command->setLocalizedMetaDescriptions($editableRootCategoryTestCaseData->getMetaDescription());
-        $command->setLocalizedMetaKeywords($editableRootCategoryTestCaseData->getMetaKeywords());
-        $command->setAssociatedGroupIds($editableRootCategoryTestCaseData->getGroupAssociationIds());
+        if (isset($data['description'])) {
+            $command->setLocalizedDescriptions($data['description']);
+        }
+        if (isset($data['meta description'])) {
+            $command->setLocalizedMetaDescriptions($data['meta description']);
+        }
+        if (isset($data['meta title'])) {
+            $command->setLocalizedMetaTitles($data['meta title']);
+        }
+        if (isset($data['active'])) {
+            $command->setIsActive(PrimitiveUtils::castStringBooleanIntoBoolean($data['active']));
+        }
+        if (isset($data['additional description'])) {
+            $command->setLocalizedAdditionalDescriptions($data['additional description']);
+        }
+        if (isset($data['group access'])) {
+            $command->setAssociatedGroupIds($this->referencesToIds($data['group access']));
+        }
+        if (isset($data['associated shops'])) {
+            $command->setAssociatedShopIds($this->referencesToIds($data['associated shops']));
+        }
+        if (isset($data['meta keywords'])) {
+            $command->setLocalizedMetaKeywords($data['meta keywords']);
+        }
 
-        $this->getCommandBus()->handle($command);
+        /** @var CategoryId $categoryId */
+        $categoryId = $this->getCommandBus()->handle($command);
+        SharedStorage::getStorage()->set($categoryReference, $categoryId->getValue());
     }
 
     /**
-     * @When I add new root category :categoryReference with following details:
+     * @param EditCategoryCommand|EditRootCategoryCommand $command
+     * @param TableNode $tableNode
      *
-     * @param string $categoryReference
-     * @param TableNode $table
+     * @see EditCategoryCommand
+     * @see EditRootCategoryCommand
+     *
+     * Technically both commands should be filled separately,
+     * but it happens to match almost all properties except $parentId, so we can reuse them here.
+     *
+     * Probably these commands are implemented wrong and instead should have extended each other if they supposed to share the properties and the logic,
```

Comment:
Actually, I believe extending commands is a bad practice, each command is supposed to be independent and should define its own clear contract. Because even if it's done with the idea of avoiding duplication in mind, from a technical point of view it creates a coupling between the two commands, modifying the common parent will impact both commands even if the new field is not related to both. You can always stop sharing the common parent at that point, but I think that's why it should be avoided.

I guess only in very rare use cases can it be acceptable, for example in the currency we used it because Unofficial currencies have the exact same structure as official currencies, they just obey different domain rules which are handled by the Handler itself depending on which command you use. But even then I struggled a long time to decide if I should use a common parent or not, and I still have doubts about it 😅

Reply:
Yes it depends on what we expect. If we expect these commands going separate ways, then yes, we shouldn't have a shared contract for them.
> for example in the currency we used it because Unofficial currencies have the exact same structure as official
> currencies, they just obey different domain rules which are handled by the Handler

It seems very similar concept here to me and quite a simple example of inheritence. The "home" category is still a category, the only difference, that we don't want to manipulate its parentId, everything else seems to be the same. (its 90% chance they will both evolve together) :man_shrugging: Ofc its debatable and it is already done so nvm... Even the basic inheritence examples seems ok at first, but something new comes up and you know you failed - like `car extends vehicle - it has wheels, steering wheel, seats. Everything works and seems nice when one day there comes some tesla tank which has tracks instead of wheels and is self-driven by AI so there is neither seats nor steering wheel `:sweat_smile: ).
Anyway, I'll adjust the comment abit, but I will leave the "reusable" method here as I think for tests its ok. And its easy to separate it to dedicated methods for each command later if we see that it gets too messy.